### PR TITLE
Fix broken tests

### PR DESCRIPTION
**Pull request by shpran**
**Changes:**
* Broken tests where refactored and fixed.
* Method `isDependsOnPropertyCollectionChanged` fixed - it will return reversed boolean value.

### CODE REVIEW

File: models/classes/validator/PropertyChangedValidator.php [hunk 1]
```diff
@@ -40,37 +39,39 @@ public function isPropertyChanged(core_kernel_classes_Property $property, OldPro
...
-        if (null === $currentPropertyType && null === $oldPropertyType) {
+        if ($currentPropertyType === null && $oldPropertyType === null) {
             return false;
         }
 
         if (
-            null !== $currentPropertyType && null === $oldPropertyType
-            || null === $currentPropertyType && null !== $oldPropertyType
+            ($currentPropertyType !== null && $oldPropertyType === null)
+            || ($currentPropertyType === null && $oldPropertyType !== null)
```
Comment (gabrielfs7) [inline models/classes/validator/PropertyChangedValidator.php:55]: Good catch :)

File: models/classes/validator/PropertyChangedValidator.php [hunk 2]
```diff
@@ -85,7 +86,7 @@ public function isDependsOnPropertyCollectionChanged(
         core_kernel_classes_Property $property,
         OldProperty $oldProperty
     ): bool {
-        return $property->getDependsOnPropertyCollection()->isEqual(
+        return !$property->getDependsOnPropertyCollection()->isEqual(
```
Comment (gabrielfs7) [inline models/classes/validator/PropertyChangedValidator.php:89]: how did we miss that 🤕

Comment (shpran) [reply models/classes/validator/PropertyChangedValidator.php:89]: I don't know 🙁

File: models/classes/validator/PropertyChangedValidator.php [hunk 3]
```diff
@@ -40,37 +39,39 @@ public function isPropertyChanged(core_kernel_classes_Property $property, OldPro
...
+            ($currentPropertyType !== null && $oldPropertyType === null)
+            || ($currentPropertyType === null && $oldPropertyType !== null)
         ) {
             return true;
         }
 
-        $currentPropertyTypeUri = $currentPropertyType->getUri();
-
-        if ($currentPropertyTypeUri !== $oldPropertyType->getUri()) {
+        if ($currentPropertyType->getUri() !== $oldPropertyType->getUri()) {
```
Comment (gabrielfs7) [inline models/classes/validator/PropertyChangedValidator.php:60]: You can replace above content with:

```
return $currentPropertyType->getUri() !== $oldPropertyType->getUri();
```

File: test/unit/model/validator/PropertyChangedValidatorTest.php [hunk 4]
```diff
@@ -200,36 +320,167 @@ public function isDependsOnPropertyCollectionChangedDataProvider(): array
...
+                ->expects($this->never())
+                ->method($method);
+        } else {
+            $mockObject
+                ->expects($this->exactly($expects))
+                ->method($method)
+                ->willReturn($returnValue);
+        }
+
+        return $this;
```
Comment (gabrielfs7) [inline test/unit/model/validator/PropertyChangedValidatorTest.php:484]: No need for else:

```suggestion
        if ($expects === 0) {
            $mockObject
                ->expects($this->never())
                ->method($method);
                
            return $this;
        }
        
        $mockObject
            ->expects($this->exactly($expects))
            ->method($method)
            ->willReturn($returnValue);

        return $this;
```

File: test/unit/model/validator/PropertyChangedValidatorTest.php [hunk 5]
```diff
@@ -200,36 +320,167 @@ public function isDependsOnPropertyCollectionChangedDataProvider(): array
...
+                $expects['getDependsOnPropertyCollection'] ?? 0,
+                $dependsOnPropertyCollection
             );
+
         return $property;
     }
+
+    private function createOldPropertyMock(
+        string $propertyLabel,
+        ?core_kernel_classes_Resource  $propertyType,
```
Comment (gabrielfs7) [inline test/unit/model/validator/PropertyChangedValidatorTest.php:430]:
```suggestion
        ?core_kernel_classes_Resource $propertyType,
```

**Review by gabrielfs7 — APPROVED** [review]

File: test/unit/model/validator/PropertyChangedValidatorTest.php (hunk 4 above)
Comment (shpran) [reply test/unit/model/validator/PropertyChangedValidatorTest.php:484]: You are right, thanks 🙂

File: test/unit/model/validator/PropertyChangedValidatorTest.php (hunk 5 above)
Comment (shpran) [reply test/unit/model/validator/PropertyChangedValidatorTest.php:430]: Good catch! 😅 
Thanks!

**Review by boajer — APPROVED** [review]
That's terrific!

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**